src/light.rs

```diff
@@ -88,47 +106,65 @@ where
         })
     }
 
+    /// Read the color / ambient light sensor data.
+    ///
+    /// Blocks as long as the data is not ready.
+    /// This clears the data ready flag.
+    #[maybe_async_cfg::only_if(async)]
+    pub async fn read_light(&mut self) -> NbResult<LightData, Error<E>> {
+        while !self.is_light_data_valid().await? {}
+        let mut data = [0; 8];
+        self.read_data(Register::CDATAL, &mut data).await?;
+        Ok(LightData {
+            clear: (u16::from(data[1]) << 8) | u16::from(data[0]),
+            red: (u16::from(data[3]) << 8) | u16::from(data[2]),
+            green: (u16::from(data[5]) << 8) | u16::from(data[4]),
+            blue: (u16::from(data[7]) << 8) | u16::from(data[6]),
+        })
+    }
+
     /// Read the color / ambient light sensor clear channel data.
     ///
-    /// Returns `nb::Error::WouldBlock` as long as the data is not ready.
+    /// Blocks while the data is not ready.
```

Comment:
```suggestion
    /// Blocks (asynchronously) or returns `nb::Error::WouldBlock` while the data is not ready.
```
I think this wording would be more accurate, especially since the `nb` version does not actually block at all.
(Also applies elsewhere)

Reply:
Done.